source/new.rst

```diff
@@ -696,6 +690,80 @@ Notes on the verified group protocol
   Carol could send the key to further verified groups
   that Bob is not part of.
 
+Dealing with key loss and compromise
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If a user looses their device
+they can setup a new device
+and regain access to their inbox.
+However they may loose their secret key.
+
+They can generate a new key pair.
+Autocrypt will distribute their new public key
+in the Autocrypt headers
+and opportunistic encryption will switch to it automatically.
+
+Verified groups will remain unreadable
+until the user verifies a contact from that group.
+Then the contact can update the key used in the group.
+This happens by sending a "vg-member-setup" message
+to the group.
+Since the email address of that user remains the same
+the old key will be replaced by the new one.
+
+Implementers may decide
+wether the recipients of such key updates
+propagate them to other groups
+they share with the user in question.
+If they do this will speed up the recovery from device loss.
+However it also allows Bob-in-the-middle attacks
+that replace the originally verified keys.
+So the decision needs to be based on the threat model of the app
+and the strategy picked for verification reuse
+
+If a key is known or suspected to be compromised
+more care needs to be taken.
+Since network attackers can drop messages
+they can also drop the "vg-member-setup" message
+that was meant to replace a compromised key.
+
+Therefore an key update due to key compromise
+requires a confirmation
+by all other members of the group,
+that they received the update.
+
```

Comment:
It's precisely this implication that i am skpetical about.  It's hard to do as people are not always online etc.  How do you deal with error cases?  It's quite a different thing to have two people engage in verification where you can presume some level of onlineness.  It's a different thing to demand that group members confirm and thus are online -- what if one member out of 10 does not confirm because that person has lost their device or otherwise does not use their account currently? 

Also, your reasoning here seems to mix "key compromise" with "active network attack".  If it's "just a key compromise" and not a network attack, then confirmation is not neccessary.  If it's indeed both, then the active network attacker can not detect easily if a message is a member-setup message or a regular one.  But dropping several  messages in the primary channel leads to high chances of keyhistory-verification to trigger.

Reply:
I'm mixing the two attack vectors because verified groups are meant to protect against active network attacks. That's the threat we are designing this for. Now the question is ... how does a compromised key impact those guarantees and how can we recover from it. Saying... well maybe there is no network attack does not really cover this. If there is no network attack there's no need for a verified group. Plain AC will recover just fine.

About the online nature... I don't think the confirmation needs to happen right away. But as long as there has been no confirmation there might be mails coming in from the people that did not confirm that will be encrypted to the old key. We could see the incoming mails encrypted to the new key as confirmations... but they may as well have been reencrypted - as long as we do not have AEAD or the desired-recipients header the signature is not bound to the key used for encryption.

From my point of view the most straight forward way of dealing with key compromise is to set up a new group from scratch. If we allow verification reuse this can happen on the spot. This way it's clear that all messages in the new group are protected from MITM despite the compromised key. And all messages that still arrive to the old group are still encrypted to a compromised key.